### PR TITLE
Fix bench

### PR DESCRIPTION
**Pull request by Dieterbe**
@woodsaj can we merge this before the http refactor? (and then rebase httpRefactor). currently in master and httpRefactor the benchmarks are broken. i'ld like them to work before and after httpRefactor, so we can compare easily

### CODE REVIEW

File: mdata/aggmetric_test.go [hunk 1]
```diff
@@ -174,6 +175,10 @@ func BenchmarkAggMetrics1000Metrics1Day(b *testing.B) {
 	}
 
 	metrics := NewAggMetrics(dnstore, chunkSpan, numChunks, chunkMaxStale, metricMaxStale, ttl, 0, aggSettings)
+	go func() {
+		for range chunk.TotalPoints {
```
Comment (woodsaj) [inline mdata/aggmetric_test.go]: i really dont like that we need to do this.

File: circle.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ test:
     - cd $IMPORTPATH && go test -v -race $(go list ./... | grep -v /vendor/)
     - cd $IMPORTPATH && go vet $(go list ./... | grep -v /vendor/)
     - cd $IMPORTPATH && scripts/end2end_test.sh
+    - cd $IMPORTPATH && go test -run='^$' -bench=. $(go list ./... | grep -v /vendor/)
```
Comment (woodsaj) [inline circle.yml]: end2end_test tears down the docker containers when it finishes so all of the benchmark tests that have an external dependency will fail

File: mdata/aggmetric_test.go (hunk 1 above)
Comment (Dieterbe) [inline mdata/aggmetric_test.go]: yeah me neither actually.  now that I think about it, we could just replace the channel based approach with atomic.UpdateUint64 or something. then whether we read out the values or not doesn't really matter.  the upcoming met library also uses this approach internally, but i'ld like benchmarks to be functional before then, so i'll patch this up tomorrow I think

File: circle.yml (hunk 2 above)
Comment (Dieterbe) [inline circle.yml]: don't those benchmarks fail anyway? do you mean the cassandra-idx and elasticsearch-idx working? seems like those require extra work (e.g. spinning up those services) to get working in the first place so i never really ran them.

Comment (woodsaj) [inline circle.yml]: if you try and run the benchmarks in circleci and they fail, the circleci job will fail.
eg. https://circleci.com/gh/raintank/metrictank/1166?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Pull request merged**